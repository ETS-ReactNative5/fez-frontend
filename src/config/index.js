export {
    API_URL,
    APP_URL,
    AUTH_URL_LOGIN,
    AUTH_URL_LOGOUT,
    SESSION_COOKIE_NAME,
    TOKEN_NAME,
    ORCID_BASE_URL,
    ORCID_CLIENT_ID,
    ORCID_AUTHORIZATION_URL,
    FACET_TYPE_FILTER,
    FACET_TYPE_RANGE,
    FEZ_URL
} from './general';

export {api, generateCancelToken, cache} from './axios';
export {default as theme} from './theme';
export * as validation from './validation';
export * as general from './general';
export {publicationTypes} from './general';
export * as routes from './routes';
