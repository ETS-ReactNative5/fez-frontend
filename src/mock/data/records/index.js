export { default as publicationTypeListAudio } from './publicationTypeListAudio';
export { default as publicationTypeListBook } from './publicationTypeListBook';
export { default as publicationTypeListBookChapter } from './publicationTypeListBookChapter';
export { default as publicationTypeListConferencePaper } from './publicationTypeListConferencePaper';
export { default as publicationTypeListConferenceProceedings } from './publicationTypeListConferenceProceedings';
export { default as publicationTypeListCreativeWork } from './publicationTypeListCreativeWork';
export { default as publicationTypeListDepartmentTechnicalReport } from './publicationTypeListDepartmentTechnicalReport';
export { default as publicationTypeListImage } from './publicationTypeListImage';
export { default as publicationTypeListJournal } from './publicationTypeListJournal';
export { default as publicationTypeListManuscript } from './publicationTypeListManuscript';
export { default as publicationTypeListNewspaperArticle } from './publicationTypeListNewspaperArticle';
export { default as publicationTypeListPatent } from './publicationTypeListPatent';
export { default as publicationTypeListPreprint } from './publicationTypeListPreprint';
export { default as publicationTypeListResearchReport } from './publicationTypeListResearchReport';
export { default as publicationTypeListSeminarPaper } from './publicationTypeListSeminarPaper';
export { default as publicationTypeListThesis } from './publicationTypeListThesis';
export { default as publicationTypeListVideo } from './publicationTypeListVideo';
export { default as publicationTypeListWorkingPaper } from './publicationTypeListWorkingPaper';
export { default as collectionRecord } from './collectionRecord';
export { default as collections } from './collections';
export { default as collectionsByCommunity } from './collectionsByCommunity';
export { default as communityRecord } from './communityRecord';
export { default as communitySearchList } from './communitySearchList';
export { default as externalDoiSearchResultList } from './externalDoiSearchResultList';
export { default as externalPubMedSearchResultsList } from './externalPubMedSearchResultsList';
export { default as externalTitleScopusResultsList } from './externalTitleScopusResultsList';
export { default as externalTitleSearchResultsList } from './externalTitleSearchResultsList';
export { default as incompleteNTROlist } from './incompleteNTROlist';
export { default as incompleteNTROrecord } from './incompleteNTROrecord';
export { default as incompleteNTROrecordUqrdav10 } from './incompleteNTRORecordUqrdav10';
export { default as incompleteNTROrecordUqsbutl1 } from './incompleteNTRORecordUqsbutl1';
export { default as internalTitleSearchList } from './internalTitleSearchList';
export { default as internalTitleSearchListNoResults } from './internalTitleSearchListNoResults';
export { default as lookupToolIncites } from './lookupToolIncites';
export { default as MyDatasetList } from './MyDatasetList';
export { default as myRecordsList } from './myRecordsList';
export { default as possibleUnclaimedList } from './possibleUnclaimedList';
export { default as record } from './record';
export { default as recordsTypeList } from './recordsTypeList';
export { default as recordWithDatastreams } from './recordWithDatastreams';
export { default as recordWithoutAuthorIds } from './recordWithoutAuthorIds';
export { default as recordWithTiffAndThumbnail } from './recordWithTiffAndThumbnail';
export { default as UQ352045 } from './incompleteNTRORecordUQ352045';
export { default as UQ716942uqagrinb } from './incompleteNTRORecordUQ716942_uqagrinb';
export { default as UQ716942uqagrinbGrants } from './incompleteNTRORecordUQ716942_uqagrinb_grants';
