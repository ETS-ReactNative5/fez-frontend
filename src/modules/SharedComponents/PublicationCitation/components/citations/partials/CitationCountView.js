import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {ExternalLink} from 'modules/SharedComponents/ExternalLink';

export default class CitationCountView extends PureComponent {
    static propTypes = {
        source: PropTypes.string,
        count: PropTypes.any,
        link: PropTypes.string,
        title: PropTypes.string
    }

    constructor(props) {
        super(props);
    }

    render() {
        const {source, count, link, title} = this.props;
        return (
            <React.Fragment>
                <ExternalLink
                    className={`${source}CitationCount`}
                    href={link}
                    aria-label={title}
                    title={title}
                    openInNewIcon={false}>
                    <span className={`fez-icon ${source} large`}/>
                    &nbsp;
                    <span className="citationCountNumber">{count}</span>
                    &nbsp;&nbsp;
                </ExternalLink>
            </React.Fragment>
        );
    }
}
