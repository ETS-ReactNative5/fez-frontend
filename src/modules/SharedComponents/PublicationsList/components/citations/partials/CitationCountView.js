import React from 'react';
import PropTypes from 'prop-types';
import {ExternalLink} from 'modules/SharedComponents/ExternalLink';

const CitationCountView = ({source, count, link, title}) => {
    return (
        <span className="citationCount">
            <ExternalLink
                className={`${source}CitationCount`}
                href={link}
                aria-label={title}
                title={title}
                openInNewIcon={false}
            >
                <div className={`fez-icon ${source} large`} />
                <span className="citationCountNumber">{count}</span>
            </ExternalLink>
        </span>
    );
};

CitationCountView.propTypes = {
    source: PropTypes.string,
    count: PropTypes.number,
    link: PropTypes.string,
    title: PropTypes.string
};

export default CitationCountView;
