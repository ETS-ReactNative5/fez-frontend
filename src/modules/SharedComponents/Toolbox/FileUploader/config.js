export const CLOSED_ACCESS_ID = 8;
export const OPEN_ACCESS_ID = 9;
export const SIZE_BASE = 1000;

export const DEFAULT_FILE_UPLOAD_LIMIT = 10;
export const DEFAULT_MAX_FILE_SIZE = 8;

export const SIZE_UNIT_B = 'B';
export const SIZE_UNIT_K = 'K';
export const SIZE_UNIT_M = 'M';
export const SIZE_UNIT_G = 'G';

export const FILE_META_KEY_ACCESS_CONDITION = 'access_condition_id';
export const FILE_META_KEY_EMBARGO_DATE = 'date';

export const SIZE_UNITS = [SIZE_UNIT_B, SIZE_UNIT_K, SIZE_UNIT_M, SIZE_UNIT_G];

export const FILE_NAME_RESTRICTION = /^(?=^\S*$)(?=^[a-z\d\-_]+(\.\d{3}|\.r\d{2,3}|\.part\d{1,3})?\.[^\.]+$)(?=.{1,45}$)(?!(web_|preview_|thumbnail_|stream_|fezacml_|presmd_|\d))[a-z\d\-_\.]+/;

/**
 * Note: ext -> mimetype is a 1-n relation, but we only care about one value
 */
export const MIME_TYPE_WHITELIST = {
    pdf: 'application/pdf',
    jpe: 'image/jpeg',
    jpeg: 'image/jpeg',
    jpg: 'image/jpeg',
    gif: 'image/gif',
    tif: 'image/tiff',
    tiff: 'image/tiff',
    png: 'image/png',
    zip: 'application/zip',
    tar: 'application/x-tar',
    gz: 'application/gzip',
    '7z': 'application/x-7z-compressed',
    rar: 'application/x-rar-compressed',
    mk3d: 'video/x-matroska',
    mks: 'video/x-matroska',
    mkv: 'video/x-matroska',
    qt: 'video/quicktime',
    mov: 'video/quicktime',
    mp4: 'video/mp4',
    mp4v: 'video/mp4',
    mp3: 'audio/mp3',
    wav: 'audio/wav',
    mpe: 'video/mpeg',
    mpeg: 'video/mpeg',
    mpg: 'video/mpeg',
    m1v: 'video/mpeg',
    m2v: 'video/mpeg',
    wmv: 'video/x-ms-wmv',
    wma: 'audio/x-ms-wma',
    avi: 'video/x-msvideo',
    m4a: 'audio/x-m4a',
    csv: 'text/csv',
    xla: 'application/vnd.ms-excel',
    xlc: 'application/vnd.ms-excel',
    xlm: 'application/vnd.ms-excel',
    xls: 'application/vnd.ms-excel',
    xlt: 'application/vnd.ms-excel',
    xlw: 'application/vnd.ms-excel',
    xlsx: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
    ods: 'application/vnd.oasis.opendocument.spreadsheet',
    gsheet: 'application/vnd.google-apps.spreadsheet',
};
