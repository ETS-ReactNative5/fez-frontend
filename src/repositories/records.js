import {post, patch} from './generic';
import * as routes from './routes';

/**
 * Posts a records item
 * @param {object} data to be posted, refer to backend API
 * @returns {Promise}
 */
export function postRecord(data) {
    if (data.rek_pid) return Promise.resolve(data);
    return post(routes.NEW_RECORD_API(), data);
}

/**
 * Patches a records item
 * @param {string} pid of object to be updated
 * @param {object} data to be posted, refer to backend API
 * @returns {Promise}
 */
export function patchRecord(pid, data) {
    return patch(routes.EXISTING_RECORD_API({pid: pid}), data);
}
