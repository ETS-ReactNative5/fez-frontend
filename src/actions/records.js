import {post, patch} from 'repositories/generic';
import * as routes from 'repositories/routes';
import {putUploadFiles} from '../repositories';
import * as transformers from './transformers';
import {NEW_RECORD_DEFAULT_VALUES} from 'config/general';
import * as actions from './actionTypes';

/**
 * Save a new record involves up to three steps: create a new record, upload files, update record with uploaded files.
 * If error occurs on any stage failed action is dispatched
 * @param {object} data to be posted, refer to backend API
 * @returns {promise} - this method is used by redux form onSubmit which requires Promise resolve/reject as a return
 */
export function createNewRecord(data) {
    return dispatch => {
        dispatch({type: actions.CREATE_RECORD_SAVING});

        // set default values, links
        const recordRequest = {
            ...NEW_RECORD_DEFAULT_VALUES,
            ...JSON.parse(JSON.stringify(data)),
            ...transformers.getRecordLinkSearchKey(data),
            ...transformers.getRecordAuthorsSearchKey(data.authors || data.currentAuthor),
            ...transformers.getRecordAuthorsIdSearchKey(data.authors || data.currentAuthor),
            ...transformers.getRecordContributorsSearchKey(data.editors),
            ...transformers.getRecordContributorsIdSearchKey(data.editors),
            ...transformers.getRecordSupervisorsSearchKey(data.supervisors),
            ...transformers.getRecordSubjectSearchKey(data.fieldOfResearch)
        };

        // delete extra form values from request object
        if (recordRequest.authors) delete recordRequest.authors;
        if (recordRequest.editors) delete recordRequest.editors;
        if (recordRequest.files) delete recordRequest.files;
        if (recordRequest.currentAuthor) delete recordRequest.currentAuthor;
        if (recordRequest.supervisors) delete recordRequest.supervisors;
        if (recordRequest.fieldOfResearch) delete recordRequest.fieldOfResearch;

        let newRecord = null;
        const hasFilesToUpload = data.files && data.files.queue && data.files.queue.length > 0;
        const recordPatch = hasFilesToUpload ? {...transformers.getRecordFileAttachmentSearchKey(data.files.queue)} : null;

        return post(routes.NEW_RECORD_API(), recordRequest)
            .then(response => {
                // save new record response
                newRecord = response.data;
                return response;
            })
            .then(() =>(hasFilesToUpload ? putUploadFiles(newRecord.rek_pid, data.files.queue, dispatch) : newRecord))
            .then(() => (hasFilesToUpload ? patch(routes.EXISTING_RECORD_API({pid: newRecord.rek_pid}), recordPatch) : newRecord))
            .then((response) => {
                dispatch({
                    type: actions.CREATE_RECORD_SUCCESS,
                    payload: response.data ? response.data : newRecord
                });
                return Promise.resolve(response.data ? response.data : newRecord);
            })
            .catch(error => {
                // record was created, but file upload or record patch failed
                if (!!newRecord && !!newRecord.rek_pid) {
                    dispatch({
                        type: actions.CREATE_RECORD_SUCCESS,
                        payload: {
                            newRecord: newRecord,
                            fileUploadFailed: true
                        }
                    });

                    return Promise.resolve(newRecord);
                }

                dispatch({
                    type: actions.CREATE_RECORD_FAILED,
                    payload: error.message
                });

                return Promise.reject(error);
            });
    };
}


/**
 * Submit thesis involves two steps: upload files, create record with uploaded files.
 * If error occurs on any stage failed action is dispatched
 * @param {object} data to be posted, refer to backend API
 * @returns {promise} - this method is used by redux form onSubmit which requires Promise resolve/reject as a return
 */
export function submitThesis(data, author) {
    return dispatch => {
        const hasFilesToUpload = data.files && data.files.queue && data.files.queue.length > 0;
        if (!hasFilesToUpload) {
            // reject thesis submission, files are required
            return Promise.reject('Please attach files to proceed with thesis submission');
        }
        // set default values, links
        const recordRequest = {
            ...JSON.parse(JSON.stringify(data)),
            ...transformers.getRecordAuthorsSearchKey(data.currentAuthor),
            ...transformers.getRecordAuthorsIdSearchKey(data.currentAuthor),
            ...transformers.getRecordSupervisorsSearchKey(data.supervisors),
            ...transformers.getRecordSubjectSearchKey(data.fieldOfResearch),
            ...transformers.getRecordFileAttachmentSearchKey(data.files.queue)
        };

        // delete extra form values from request object
        if (recordRequest.authors) delete recordRequest.authors;
        if (recordRequest.editors) delete recordRequest.editors;
        if (recordRequest.files) delete recordRequest.files;
        if (recordRequest.currentAuthor) delete recordRequest.currentAuthor;
        if (recordRequest.supervisors) delete recordRequest.supervisors;
        if (recordRequest.fieldOfResearch) delete recordRequest.fieldOfResearch;

        let fileUploadSucceeded = false;
        dispatch({type: actions.CREATE_RECORD_SAVING});

        // debugger;
        return putUploadFiles(`UQ:${author.aut_student_username}`, data.files.queue, dispatch)
            .then((response) => {
                fileUploadSucceeded = !!response;
                return post(routes.NEW_RECORD_API(), recordRequest);
            })
            .then(response => {
                dispatch({
                    type: actions.CREATE_RECORD_SUCCESS,
                    payload: {
                        newRecord: response
                    }
                });
                return response;
            })
            .catch(error => {
                const specificError = !fileUploadSucceeded
                    ? 'File upload failed. Issue has been created to notify eSpace administrators. '
                    : 'Error occurred while saving record to eSpace. ';
                const compositeError = `${specificError} ${ error.message ? `(${error.message})` : '' }`;

                dispatch({
                    type: actions.CREATE_RECORD_FAILED,
                    payload: compositeError
                });

                return Promise.reject(compositeError);
            });
    };
}
/**
 * Clear new record
 * @returns {action}
 */
export function clearNewRecord() {
    return dispatch => {
        dispatch({
            type: actions.CREATE_RECORD_RESET
        });
    };
}
