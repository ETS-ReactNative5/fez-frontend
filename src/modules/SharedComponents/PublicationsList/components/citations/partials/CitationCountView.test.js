jest.dontMock('./CitationCountView');

import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import CitationCountView from './CitationCountView';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import PropTypes from 'prop-types';
import injectTapEventPlugin from 'react-tap-event-plugin';
import {locale} from 'config';
import {researchReport} from 'mock/data/testing/records';


function setup({source, count, link, title, isShallow = false}) {
    const props = {
        source: source,
        count: count,
        link: link,
        title: title
    };

    if(isShallow) {
        return shallow(<CitationCountView {...props} />);
    }

    return mount(<CitationCountView {...props} />, {
        context: {
            muiTheme: getMuiTheme()
        },
        childContextTypes: {
            muiTheme: PropTypes.object.isRequired
        }
    });
}

beforeAll(() => {
    injectTapEventPlugin();
});

describe('CitationCountView test ', () => {
    it('should render component with given count', () => {
        const wrapper = setup({source: 'wos', count: 4, link: 'www.google.com', title: 'Google'});
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
