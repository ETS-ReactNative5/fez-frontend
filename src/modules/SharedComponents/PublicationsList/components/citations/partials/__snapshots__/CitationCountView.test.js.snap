// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitationCountView test  should render component with given count 1`] = `
<CitationCountView
  count={4}
  link="www.google.com"
  source="wos"
  title="Google"
>
  <span
    className="citationCount"
  >
    <ExternalLink
      aria-label="Google"
      className="wosCitationCount"
      externalLinkIcon={false}
      href="www.google.com"
      rel="noopener noreferrer"
      target="_blank"
      title="Google"
    >
      <a
        aria-label="Google"
        className="wosCitationCount"
        href="www.google.com"
        rel="noopener noreferrer"
        target="_blank"
        title="Google"
      >
        <div
          className="fez-icon wos large"
        />
        <span
          className="citationCountNumber"
        >
          4
        </span>
      </a>
    </ExternalLink>
  </span>
</CitationCountView>
`;
